src/System.Private.ServiceModel/src/System/ServiceModel/Channels/WebSocketTransportDuplexSessionChannel.cs

```diff
@@ -134,6 +134,8 @@ protected override void CompleteClose(TimeSpan timeout)
             {
                 TD.WebSocketConnectionClosed(WebSocket.GetHashCode());
             }
+
+            Cleanup();
```

Comment:
I've looked into this a little more and I don't think this is the right fix. In the OnClose method, we call Cleanup() in a finally block. I think the problem is I didn't override the OnCloseAsync method which should call the Cleanup method. This means the base class  TransportDuplexSessionChannel.OnCloseAsync() is being called and we're missing the Cleanup() call.